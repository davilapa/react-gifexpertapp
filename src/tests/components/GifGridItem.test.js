import React from 'react';
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifGridItem } from '../../components/GifGridItem';
describe('Pruebas en <GifGridItem />', () => {
    const props = { title: 'Title example', url: 'test.com' };
    const wrapper = shallow(<GifGridItem {...props} />)
    test('Debe de mostrar <GifGridItem />  correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de tener un parrafo con el title', () => {
        const p = wrapper.find('p');

        expect(p.text().trim()).toBe(props.title);
    });

    test('Debe de tener la igual al url y alt de las props', () => {
        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(props.url);
        expect(img.prop('alt')).toBe(props.title);
    });

    test('Debe de tener animate__fadeIn', () => {
        const div = wrapper.find('div');
        
        expect(div.prop('className')).toContain('animate__fadeIn')
    });
} )