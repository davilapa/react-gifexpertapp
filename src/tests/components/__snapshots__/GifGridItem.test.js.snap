// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe de mostrar <GifGridItem />  correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Title example"
    src="test.com"
  />
  <p>
    Title example
  </p>
</div>
`;
